### ISSUE TITLE
ClusterData and ClusterLoader shuffling labels even when `shuffle=False`

### ISSUE THREAD

**Issue by prashantpandey** [pyg-team/pytorch_geometric#1579]
## 🐛 Bug

I am trying to use the ClusterData and ClusterLoader to partition a big graph and run a GCN. But when I output the final labels they are not in the same order as they were when I constructed the Data object. To verify this I created a node id array in the Data object and tried to print it out by looping through the ClusterLoader. 

Does ClusterData/Loader shuffle labels in the graph data?

## To Reproduce

```
n = torch.tensor(list(range(1, node_count)), dtype=torch.int)
data = Data(x=x, edge_index=edge_index, y=y, n=n)
cluster_data = ClusterData(data, num_parts=1000, recursive=False, save_dir=dataset.processed_dir)
loader = ClusterLoader(cluster_data, batch_size=20, shuffle=False, num_workers=5)

f = open('temp', 'w')
for data in loader:
    name = data.n.tolist()
    for n in name:
        f.write(str(int(n)) + '\n')
f.close()
```
I expect to see the name list numbers from 1 to `node_count` in order. 
However, the numbers are shuffled. Also, the maximum number is ~579 and then the counter gets reset. 

However, if I use a simple loader I get the correct result.
```
simple_loader = DataLoader(dataset, batch_size=5, shuffle=False, num_workers=1)
f = open('temp_s', 'w')
for data in simple_loader:
    name = data.n.tolist()
    for n in name:
        f.write(str(int(n)) + '\n')
f.close()
```

## Environment

* OS:
* Python version: Python 3.7.7
* PyTorch version: pytorch/v1.5.1-gpu

**Comment by rusty1s** [comment]
Hi and thanks for this issue,
this behavior is definitely intended, since `ClusterData` permutes the data so that nodes in the same partition are contiguous in memory.
